### ISSUE TITLE
Create new app fails on Devise

### ISSUE THREAD

**Issue by samgaw** [thredded/thredded_create_app#12]
It looks like something has changed in Devise's config that is breaking the pattern matching. Running with verbose I can see:
```
bundle exec rails g devise:i18n:views -v sessions registrations
      invoke  Devise::I18n::SharedViewsGenerator
      create    app/views/devise/shared
      create    app/views/devise/shared/_links.html.erb
      invoke  Devise::I18n::MailerViewsGenerator
      invoke  i18n:simple_form_for
      create    app/views/devise/sessions
      create    app/views/devise/sessions/new.html.erb
      create    app/views/devise/registrations
      create    app/views/devise/registrations/edit.html.erb
      create    app/views/devise/registrations/new.html.erb
thredded_create_app: No match found for (?-mix:protect_from_forgery.*\n) in app/controllers/application_controller.rb
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/tasks/base.rb:87:in `replace'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/tasks/base.rb:128:in `inject_into_file'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/tasks/add_devise.rb:33:in `setup_after_sign_in_behaviour'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/tasks/add_devise.rb:24:in `after_bundle'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/generator.rb:37:in `each'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/generator.rb:37:in `block in generate'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/generator.rb:59:in `block in in_app_env'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/generator.rb:58:in `chdir'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/generator.rb:58:in `in_app_env'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/generator.rb:33:in `generate'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/cli.rb:59:in `run'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/cli.rb:34:in `block in start'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/cli.rb:169:in `auto_output_coloring'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/cli.rb:32:in `start'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/lib/thredded_create_app/cli.rb:24:in `start'
/Users/samgaw/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thredded_create_app-0.1.22/exe/thredded_create_app:9:in `<top (required)>'
/Users/samgaw/.rbenv/versions/2.5.0/bin/thredded_create_app:23:in `load'
/Users/samgaw/.rbenv/versions/2.5.0/bin/thredded_create_app:23:in `<main>'
```

I had a look at [add_devise.rb:33](https://github.com/thredded/thredded_create_app/blob/master/lib/thredded_create_app/tasks/add_devise.rb#L32) but admittedly my knowledge of Ruby sucks and I'm not entirely sure what it's trying to insert after `protect_from_forgery.*`. Any help appreciated.

**Comment by glebm** [comment]
It's actually trying to insert into `app/controllers/application_controller.rb`.

What do you see in `app/controllers/application_controller.rb`?

**Comment by samgaw** [comment]
Not a lot :)
```ruby
$: cat app/controllers/application_controller.rb
class ApplicationController < ActionController::Base
end
```

That would probably explain why it's failing to insert after protect_from_forgery yeah?

**Comment by glebm** [comment]
Yes, but I don't understand why it's empty. The Rails template has `protect_from_forgery`.
What version of Rails does it install?

**Comment by glebm** [comment]
You can also try passing the Rails version manually: `--rails-version 5.2.0.rc1`

**Comment by samgaw** [comment]
Gemfile has `gem 'rails', '~> 5.2.0.rc1'`

If I create a new app with rails new I see:
```ruby
$: cat testing/app/controllers/application_controller.rb
class ApplicationController < ActionController::Base
end
```

**Comment by glebm** [comment]
Confirmed, this is the template in Rails 5.2.0.rc:
https://github.com/rails/rails/blob/v5.2.0.rc1/railties/lib/rails/generators/rails/app/templates/app/controllers/application_controller.rb.tt

`protect_from_forgery` was moved from the template into `ActionController::Base` in this Rails commit: https://github.com/rails/rails/commit/ec4a836919c021c0a5cf9ebeebb4db5e02104a55#diff-8380e0e11d4efeb2b0b142ae9e4cf512

The generator here will need to be adjusted

**Comment by samgaw** [comment]
Nice find. I was beginning to have serious doubts about myself...

**Comment by samgaw** [comment]
Thanks! I changed the local gem and everything is working.

**Comment by glebm** [comment]
This is now released in v0.1.23!
